lib/strings-with-deps.nix

```
@@ -52,10 +52,101 @@ let
 in
 rec {
 
-  /* !!! The interface of this function is kind of messed up, since
-     it's way too overloaded and almost but not quite computes a
-     topological sort of the depstrings. */
-
+  /**
+    Topologically sort a collection of dependent strings.
+    Only depstrings listed in `arg` and their dependencies will be included in the result.
+
+    ::: {.note}
+    Doesn't formally fulfill the topological sort definition, but it's good enough for our purpose in nixpkgs.
+    :::
+
+    # Inputs
+
+    `predefined` (attribute set)
+
+    : set of depstrings. (String or DepString)
+      A depstring is itself an attribute set with the following attributes:
+      - `deps` (list of strings)
+      - `text` (string)
+      A depstring can be a simple string if it has no dependencies.
```

This is kinda contradictory, because a DepString is explained to be `String | { deps, text }`.

```suggestion
    : strings with annotated dependencies (strings or attribute set)
      A value can be a simple string if it has no dependencies.
      Otherwise, is can be an attribute set with the following attributes:
      - `deps` (list of strings)
      - `text` (Any)
```

Note: the value can really be anything!